File: delivery-service/api/src/main/java/org/delivery/api/domain/token/service/TokenService.java
package org.delivery.api.domain.token.service;

import lombok.RequiredArgsConstructor;
import org.delivery.api.domain.token.ifs.TokenHelper;
import org.delivery.api.domain.token.model.TokenDto;
import org.delivery.common.error.ErrorCode;
import org.delivery.common.exception.ApiException;
import org.springframework.stereotype.Service;

import java.util.HashMap;
import java.util.Objects;

/*@Service
@RequiredArgsConstructor
public class TokenService {

    private final TokenHelper tokenHelper;

    public TokenDto issueAccessToken(Long userId){
        var data = new HashMap<String, Object>();
        data.put("userId", userId);
        return tokenHelper.issueAccessToken(data);
    }

    public TokenDto issueRefreshToken(Long userId){
        var data = new HashMap<String, Object>();
        data.put("userId", userId);
        return tokenHelper.issueRefreshToken(data);
    }

    public Long validationToken(String token){
        var map = tokenHelper.validationTokenWithThrow(token);
        var userId = map.get("userId");
        Objects.requireNonNull(userId, ()->{throw  new ApiException(ErrorCode.NULL_POINT);});

        return Long.parseLong(userId.toString());
    }
}*/

File: delivery-service/account/src/main/kotlin/org/delivery/account/domain/token/service/TokenService.kt
package org.delivery.account.domain.token.service

import org.delivery.account.domain.token.ifs.TokenHelper
import org.delivery.account.domain.token.model.TokenDto
import org.springframework.stereotype.Service

@Service
class TokenService(
    private val tokenHelper: TokenHelper
) {

    fun issueAccessToken(userId: Long?)
    : TokenDto?{
        return userId?.let {
            val data = mapOf("userId" to it)
            tokenHelper.issueAccessToken(data)
        }
    }

    fun issueRefreshToken(userId: Long?)
    : TokenDto?{
        requireNotNull(userId)
        val data = mapOf("userId" to userId)
        return tokenHelper.issueRefreshToken(data)
    }

    fun validationToken(token: String)
    : Long?{
        /*
        requireNotNull(token)

        val map = tokenHelper.validationTokenWithThrow(token)

        val userId = map?.get("userId")

        requireNotNull(userId)

        return userId.toString().toLong()
        */

        //이렇게 처리도 가능
        return token?.let { token ->
            tokenHelper.validationTokenWithThrow(token)
            ?.let { map ->
                map["userId"]
            }?.let { userId ->
                userId.toString().toLong()
            }
        }
    }

}